src/MUIDataTable.js

```diff
@@ -196,8 +196,16 @@ class MUIDataTable extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.data !== prevProps.data || this.props.columns !== prevProps.columns) {
-      this.setTableData(this.props, TABLE_LOAD.INITIAL, () => {
+    if (
+      this.props.data !== prevProps.data ||
+      this.props.columns !== prevProps.columns ||
+      this.props.options !== prevProps.options
+    ) {
+      if (this.props.options !== prevProps.options) {
+        this.getDefaultOptions(this.props);
```

Comment:
Reverting back to the default props here is problematic as it conflicts with changes in state. For example, this opens a regression where setting a page number in props and then changing the page number in the table causes any change in options to revert back to the original page.